en_US/advanced/retained.md

```diff
@@ -8,18 +8,32 @@ When a client establishes a subscription, and there are retained messages with m
 
+## Flow control
+
+The message read and deliver flow rate control. When a client subscribes to a wildcard topic, many retained messages may be loaded. If you don't want these message to be loaded into the memory all at once, you can use `retainer.flow_control` options to control this behaviour. The processing flow:
+```
+load batch_read_number retained message from storage ->
+  deliver batch_deliver_number of messages ->
+  repeat this, until all retianed messages are delivered
+```
+
 ## Configuration
 
 EMQX Broker's retained message feature is implemented by the `emqx_retainer` plugin, which is enabled by default. By modifying the configuration of the ` emqx_retainer` plugin, you can adjust the storage location of the retained messages in EMQX Broker, adjust the number of preserved retained messages and the maximum length of Payload, and adjust the expiration time of retained messages. For more information about the EMQX Broker plug-in, see [plugin](./plugins.md).
 
 The `emqx_retainer` plugin is enabled by default, and the configuration path of the plugin is `etc/plugins/emqx_retainer.conf`.
 
-| Configuration item       | Type  | Optional value      | Default value | Description                                               |
-| ------------------------------ | -------- | ------------------------ | ------ | ------------------------------------------------------------ |
-| retainer.storage_type          | enum     | `ram`, `disc`, `disc_only` | ram |ram: only stored in memory; <br /> disc: stored in memory and hard disk; <br /> disc_only: only stored in hard disk|
-| retainer.max_retained_messages | integer  | \>= 0                    | 1      | The maximum number of retained messages, and 0 means no limit. After the number of retained messages exceeds the maximum limit, you can replace the existing retained messages, but cannot store retained messages for new topics. |
-| retainer.max_payload_size      | bytesize |                          | 1MB    | Retain the maximum Payload value of the message. After the Payload value exceeds the maximum value, the EMQX broker will treat the retained reserved message as a normal message. |
-| retainer.expiry_interval       | duration |                          | ０     | The expiration time of retaining message, and 0 means never expire. If the message expiration interval is set in the PUBLISH packet, the message expiration interval in the PUBLISH packet shall prevail. |
+| Configuration item                         | Type     | Optional value             | Default value | Description                                                                                                                                                                                                                        |
```

Comment:
It looks like "Optional value" would be more like "Possible values"?  "Optional value" gives me the impression that this column would have only true/false values to indicate if the line is optional.  :thinking:

Reply:
```suggestion
| Configuration item                         | Type     | Possible values             | Default value | Description                                                                                                                                                                                                                        |
```